examples/svm/plot_svm_scale_c.py

```diff
@@ -67,26 +60,49 @@
 )
 
 # %%
-# Now, we can define a linear SVC with the `l1` penalty.
+# L1-penalty case
+# ---------------
+# In the L1 case, theory says that prediction consistency (i.e. that under
+# given hypothesis, the estimator learned predicts as well as a model knowing
+# the true distribution) is not possible because of the bias of the L1. It
+# does say, however, that model consistency, in terms of finding the right set
+# of non-zero parameters as well as their signs, can be achieved by scaling
+# `C`.
```

Comment:
I am not exactly sure what to do with this example. I am not sure to understand the claim here.

In the code below, I wanted to test with a `LogisticRegression` and use a proper scoring rule (i.e. `neg_log_loss`) and I obtain a similar graph:

![image](https://github.com/scikit-learn/scikit-learn/assets/7454015/f6c1f571-a059-4528-ba02-beacd7addeb6)

Scaling `C` will only impact the lower bound for which we should not try lower `C` values. By scaling `C`, the lower bound is the same independent of the number of samples. However, when it comes to the max score, the optimal `C` is the same only in the case where we don't scale `C`. By scaling `C`, we shift the optimal value which becomes dependent on the number of samples. Thus, it seems to be the opposite of what we intend to explain.

@agramfort @mathurinm Since you are knowing more in detail about L1 linear models, do you have an idea where am I wrong here?

Reply:
I just need to run this file in the PR to get this?